exasol/toolbox/nox/_dependencies.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import annotations
+import tomlkit
+import sys
+from pathlib import Path
+import nox
+from nox import Session
+from noxconfig import PROJECT_CONFIG
+from typing import (
+    List,
+    Dict
+)
+import rich.console
+
+
+@nox.session(name="dependencies-check", python=False)
+def dependency_check(session: Session) -> None:
+    content = Path(PROJECT_CONFIG.root, "pyproject.toml").read_text()
+    dependencies = Dependencies.parse(content)
+    console = rich.console.Console()
+    if illegal := dependencies.illegal:
+        report_illegal(illegal, console)
+        sys.exit(1)
+
+
+class Dependencies:
+    def __init__(self, illegal: Dict[str, List[str]] | None):
+        self._illegal = illegal or {}
+    @staticmethod
+    def parse(pyproject_toml: str) -> "Dependencies":
+        def _source_filter(version) -> bool:
+            ILLEGAL_SPECIFIERS = ['url', 'git', 'path']
+            return any(
+                specifier in version
+                for specifier in ILLEGAL_SPECIFIERS
+            )
+
+        def _extract_dependencies(section) -> List[str]:
+            dependencies = []
+            for name, version in section.items():
+                if _source_filter(version):
+                    dependencies.append(f"{name} = {version}")
+            return dependencies
+        illegal: Dict[str, List[str]] = {}
+        toml = tomlkit.loads(pyproject_toml)
+        poetry = toml.get("tool", {}).get("poetry", {})
+        part = poetry.get("dependencies", {})
+        dependencies_list = _extract_dependencies(part)
```

Comment:
I think there are some repetitions (DRY = Don't repeat yourself):
* `part = ...`
* `list = extract()`
* `if list ...`

My recommendation would be 
1. use identical name `section` as in find_illegal() rather than introducing a different name `part`
2. use the walruss operator in the if statement: `if group := find_illegal(section):`